src/librustc_resolve/lib.rs

```diff
@@ -4254,35 +4230,14 @@ impl<'a> Resolver<'a> {
                                 resolve_error(
                                     self,
                                     span,
-                                    ResolutionError::ParametersFromOuterFunction(
-                                        def,
-                                        ParamKindOrd::Type,
-                                    ),
+                                    ResolutionError::TypeParametersFromOuterFunction(def),
                                 );
                             }
                             return Def::Err;
                         }
                     }
                 }
             }
-            Def::ConstParam(..) => {
-                for rib in ribs {
-                    if let ItemRibKind = rib.kind {
-                        // This was an attempt to use a const parameter outside its scope.
-                        if record_used {
-                            resolve_error(
-                                self,
-                                span,
-                                ResolutionError::ParametersFromOuterFunction(
-                                    def,
-                                    ParamKindOrd::Const,
-                                ),
-                            );
-                        }
-                        return Def::Err;
-                    }
-                }
-            }
```

Comment:
Looks like this was reverted incorrectly, unlike `TyParamAsConstParamTy` the `ParametersFromOuterFunction` part is still needed.

Reply:
I'm not sure we want something as strict as `ParametersFromOuterFunction`, though, because function bodies should be able to refer to the const generic parameters. For example:
```rust
fn foo<const X: u32>() -> u32 {
	X
}
```

Reply:
Function bodies are able to refer to type parameters though
```rust
fn check<T>() {
    let _: T;
}
```
, so I'd expect the resolve logic to be analogous.

Reply:
It doesn't seem to fall out as nicely as with types, because function items push a new `ItemRibKind` value namespace rib, which is what causes the `ParametersFromOuterFunction` error, but doesn't touch the type namespace rib, so when a const param is seem inside the body, it hits the error, but a type is correspondingly fine.

I'll try to think of another way around it.